mlflow/transformers/__init__.py

```diff
@@ -1334,16 +1351,23 @@ def _extract_torch_dtype_if_set(pipeline):
         return str(torch_dtype)
 
 
-def _get_or_infer_task_type(model, task: Optional[str] = None) -> str:
+def _get_or_infer_task_type(model, task: Optional[str] = None) -> Tuple[str, str]:
     """
-    Validates that a supplied task type is supported by the ``transformers`` library if supplied,
-    else, if not supplied, infers the appropriate task type based on the model type.
+    Determines whether a ``transformers`` task type or a MLflow inference task type is provided.
+    If ``transformers`` task type is not provided, the appropriate ``transformers`` task type
+    based on the model type.
     """
-    if task:
-        _validate_transformers_task_type(task)
+    transformers_task, llm_inference_task = None, None
+
+    if not task:
+        transformers_task = _infer_transformers_task_type(model)
+    elif _is_transformers_task_type(task):
+        transformers_task = task
     else:
-        task = _infer_transformers_task_type(model)
-    return task
+        transformers_task = _infer_transformers_task_type(model)
+        llm_inference_task = task
```

Comment:
I feel like this function is super confusing... hard to read.
Could you update the docstring to make it more clear?
BTW could we move up _validate_llm_inference_task_type here?